include/pybind11/iostream.h

```diff
@@ -52,9 +52,12 @@ class pythonbuf : public std::streambuf {
 
                 pywrite(line);
                 pyflush();
+
+                // Placed inside gil_scoped_aquire as a makeshift mutex
```

Comment:
Is "makeshift" the right word? Doesn't it just use the OS primitives, somewhere in the depths of CPython?

Reply:
I almost dropped it... Probably should have. If it builds, I'll drop it and we give it a second chance to fail. ;)